### PR TITLE
ci: change to use GitHub actions

### PR DESCRIPTION
**Pull request by FrankYang0529**
#### Which issue(s) this PR fixes:
https://github.com/longhorn/longhorn/issues/8039

#### Additional documentation or context

Sample build: https://github.com/FrankYang0529/longhorn-instance-manager/actions/runs/8892845609

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+    - name: Upload binaries
+      uses: actions/upload-artifact@v4
+      with:
+        name: binaries_amd64_artifact
+        path: ./bin/*
+
+  build-arm64-binaries:
+    name: Build ARM64 binaries
+    runs-on: ["self-hosted", "ARM64"]
```
Comment (innobead) [inline .github/workflows/build.yml:36]: Use label `oracle-aarch64-2cpu-8gb` instead.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        push: true
+        platforms: linux/amd64
+        tags: longhornio/longhorn-instance-manager:${{ github.ref_name }}-amd64
+        file: package/Dockerfile
+        build-args: |
+          ARCH=amd64
+
+  build-push-arm64-images:
+    name: Build and push ARM64 images
+    runs-on: ["self-hosted", "ARM64"]
```
Comment (innobead) [inline .github/workflows/build.yml:115]: Fix the label.

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, but need to fix the arm64 runner label first.

File: .github/workflows/build.yml (hunk 1 above)
Comment (FrankYang0529) [reply .github/workflows/build.yml:36]: I think we have to deploy ARM runner ourselves. The `oracle-aarch64-2cpu-8gb` can only be used in branch head. If we set it, the pull requests can't really run CI. For example, I used `oracle-aarch64-2cpu-8gb` in https://github.com/longhorn/go-spdk-helper/pull/105, but the job doesn't run.

Comment (FrankYang0529) [reply .github/workflows/build.yml]: Update to use CNCF runners. Thanks.

**Review by innobead — APPROVED** [review]

**Comment by FrankYang0529** [comment]
@mergify backport v1.6.x

**Comment by FrankYang0529** [comment]
@mergify backport v1.5.x

**Pull request merged**